components/intl/DateFromNowHourPrecision.tsx

```diff
@@ -21,36 +21,20 @@ import { differenceInHours } from 'date-fns';
 import * as React from 'react';
 import { DateSource } from 'react-intl';
 import { translate } from '../../helpers/l10n';
-import Tooltip from '../controls/Tooltip';
 import DateFromNow from './DateFromNow';
-import DateTimeFormatter from './DateTimeFormatter';
 
 export interface DateFromNowHourPrecisionProps {
   children?: (formattedDate: string) => React.ReactNode;
   date?: DateSource;
 }
 
 export default function DateFromNowHourPrecision(props: DateFromNowHourPrecisionProps) {
-  const { children = (f: string) => f, date } = props;
+  const { children: originalChildren = (f: string) => f, date } = props;
+  let children = originalChildren;
 
-  /*
-   * Cast returns as ReactElement to bypass typescript issue with functional components
-   * returning anything else (like a string, in our case...)
-   * (https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20544)
-   */
-  if (!date) {
-    return children(translate('never')) as React.ReactElement;
+  if (date && differenceInHours(Date.now(), date) < 1) {
+    children = () => originalChildren(translate('less_than_1_hour_ago'));
```

Comment:
I think this is kind of complicated.
If this triggers, then `DateFromNow` gets a children callback that ignores its param, so basically it will go through all the date formatting code for nothing.
The only benefit is the recycling of the structure, but that's just `<span title={formattedDate}>`

Why not make the hour precision a prop of `DateFromNow`, and add this condition to it?

Reply:
I agree it's quite complicated, I think even duplicating the span title thing from `DateFromNow` is fine in this case, just returning:
```ts
    <DateTimeFormatter date={parseDate(date)}>
      {(formattedDate) => (
        <span title={formattedDate}>
          {translate('less_than_1_hour_ago')}
        </span>
      )}
    </DateTimeFormatter>
```

Reply:
@gregaubert I guess you're fine with dropping the `DateFromNowHourPrecision` component then?

Reply:
I'm not sure what you mean, I think I was talking about `DateFromNowHourPrecision` implem in my comment.

Reply:
Also there's another benefit here: the message is passed to the children so it can be processed for example. Still I agree this is complicated. I'll try to simplify it.

Reply:
```ts
<DateFromNow date={user.lastConnectionDate} hourPrecision={true}>
    {date => "TOTO:" + date}
</DateFromNow>
```

wouldn't work anymore in case of _never_ or _< 1 hour_

Reply:
> I'm not sure what you mean, I think I was talking about DateFromNowHourPrecision implem in my comment.

I was referring to @jeremy-davis-sonarsource comment:

> Why not make the hour precision a prop of DateFromNow

Reply:
Ahh ok got it! Yeah I'm fine with it!